### PR TITLE
support ingest of s3:// urls

### PR DESCRIPTION
**Pull request by mhiley**
This is an implementation of #187 (support ingest of private/requester pays S3 buckets). If a url starts with `s3://`, it will use aws-sdk to download the file contents instead of failing due to unrecognized protocol.

Note, I moved aws-sdk into `dependencies` from `devDependencies` for api-lib.

This will support requester pays buckets if you set the environment variable: 
```
AWS_REQUEST_PAYER=requester
```

I tested this on my end with some simple test scripts but haven't implemented a proper sat-api test yet. I'm willing to do that but I'll need some guidance on what your expectations would be there!

Thanks 😃

### CODE REVIEW

**Comment by matthewhanson** [comment]
@mhiley thanks for the PR!  
This looks pretty good to me, but I'd also like to have @sharkinsspatial have a look.

Adding in tests would be good.  Take a look at the existing tests and see if that provides any help for what might be testable in your new code.   I'd start with `getS3ParamsFromUrl` and test out that the return value is an expected value.

The `getS3Object` function is a little trickier, as I think you'll want to mock the actual AWS call, but maybe @sharkinsspatial can weight in here too, I'm just looking into adding some new tests myself but I'm not familiar yet with them.

File: packages/api-lib/libs/ingest.js [hunk 1]
```diff
@@ -32,6 +33,49 @@ function getChildLinks(node) {
...
+    Key: key
+  }
+}
+
+function getS3Object(url) {
+  // from https://github.com/aws/aws-sdk-js/issues/1436
+  return new Promise((resolve, reject) => {
+    const s3 = new AWS.S3()
+    s3.getObject(
+      getS3ParamsFromUrl(url),
```
Comment (sharkinsspatial) [inline packages/api-lib/libs/ingest.js]: I haven't actually tested this, but it appears that in this case if `getS3ParametersFromUrl` throws an error it won't be caught and execution will stop.  I'd recommend moving the `getS3ParamsFromUrl` out of the `s3.getObject` call and reject the containing promise if an error is thrown.

**Comment by mhiley** [comment]
Thanks @matthewhanson and @sharkinsspatial - I made the requested change to reject the promise in `getS3Object` on an invalid url, and also added the requested test coverage.

I hit a snag in that sat-api's version of Ava doesn't appear to support the `throwsAsync` function, which looks like the best way to test a promise that is expected to reject. I think what I have instead in the `'getS3Object rejects promise on invalid s3 url'` test is sufficient, but let me know if there's a better way to do that.

**Comment by sharkinsspatial** [comment]
@mhiley Again, sorry for the delay.  @matthewhanson and I have both been away at conferences last week.  Can you change the base branch for this PR to `develop` rather than `master` and I will try to merge.

**Pull request merged**